src/Artifacts.jl

```diff
@@ -0,0 +1,594 @@
+#   - remove_artifact(hash): deletes `artifact_path(hash)` if it exists, does nothing otherwise.
+#   - verify_artifact(hash): verifies that the artifact maintains integrity on-disk.
+#   - archive_artifact(hash, out_path): compresses an artifact into a tarball.
+#
+# - Name level:
+#   - artifact_meta(name, toml_path; platform): returns a `Dict` of the metadata stored
+#     within a given `Artifacts.toml` file, including things like URL, platform, etc...
+#     If the artifact is a multi-map due to being platform-specific, chooses the best
+#     match among its options. Returns `nothing` if no appropriate mapping exists.
+#   - artifact_hash(args...; kwargs...): returns the hash of the requested name mapping
+#     via `artifact_meta(args...; kwargs...)["git-tree-sha1"]`, or similar.
+#   - bind_artifact(name, hash, toml_path; platform): builds a mapping from `name` to
+#     `hash` within the given `Artifacts.toml` file.  Optionally sets this as a platform-
+#     specific artifact.
+#   - unbind_artifact(name, toml_path; platform): deletes a mapping previously bound.
+#   - download_artifact(hash, url, tarball_hash): downloads an artifact from a given URL
+#     to the artifact store.
+#   - ensure_artifact_installed(name, toml_path; platform): Wrapper around
+#     `download_artifact()`, does the lookups and loop around multiple download URLs.
+#
+# - Utilities:
+#   - find_artifacts_toml(path): Attempts to find the Artifacts.toml that exists for a
+#     given source path.  Returns `nothing` if none could be found.
+
+
+const ARTIFACT_DIR_OVERRIDE = Ref{Union{String,Nothing}}(nothing)
+"""
+    with_artifacts_directory(f::Function, artifacts_dir::String)
+
+Helper function to allow temporarily changing the artifact installation directory.
+"""
+function with_artifacts_directory(f::Function, artifacts_dir::String)
+    try
+        ARTIFACT_DIR_OVERRIDE[] = artifacts_dir
+        f()
+    finally
+        ARTIFACT_DIR_OVERRIDE[] = nothing
+    end
+end
+
+function artifacts_dir(args...)
+    if ARTIFACT_DIR_OVERRIDE[] === nothing
+        return abspath(depots1(), "artifacts", args...)
+    else
+        return abspath(ARTIFACT_DIR_OVERRIDE[], args...)
+    end
+end
+
+"""
+    create_artifact(f::Function)
+
+Creates a new artifact by running `f(artifact_path)`, hashing the result, and moving it
+to the artifact store (`~/.julia/artifacts` on a typical installation).  Returns the
+identifying tree hash of this artifact.
+"""
+function create_artifact(f::Function)
+    # Ensure the `artifacts` directory exists.
+    mkpath(artifacts_dir())
+
+    # Temporary directory where we'll do our creation business
+    temp_dir = mktempdir(artifacts_dir(); prefix="create_")
+
+    try
+        # allow the user to do their work inside the temporary directory
+        f(temp_dir)
+
+        # Calculate the tree hash for this temporary directory
+        artifact_hash = SHA1(tree_hash(temp_dir))
```

Comment:
```suggestion
        artifact_hash::SHA1 = (tree_hash(temp_dir))
```

`tree_hash` already reaturns a `SHA1`

Reply:
Addressed comment above.